src/main/java/ro/utcn/sd/agui/a1/controller/ConsoleController.java

```
@@ -0,0 +1,432 @@
+        System.out.println("Text: ");
+        String text = scanner.nextLine();
+        System.out.println("Write your tags separated by blank space: ");
+        String tagNames = scanner.nextLine();
+        questionManagementService.addQuestion(currentUser.getUserId(), title, text, tagNames);
+
+
+    }
+
+    private void handleFilterQuestionsByTag() {
+
+
+        System.out.println("Write a tag: ");
+        String tagName = scanner.nextLine();
+
+        Tag foundTag = tagManagementService.findTagByName(tagName);
+        questionManagementService.filterQuestionsByTag(foundTag).forEach(s -> System.out.println(s.toString()));
+
+
+    }
+
+    private void handleFilterQuestionsByTitle() {
+
+
+        System.out.println("Write the text: ");
+        String text = scanner.nextLine();
+        questionManagementService.filterQuestionsByTitle(text).forEach(s -> System.out.println(s.toString()));
+
+    }
+
+    private void handleAddAnswer() {
+        System.out.println("Write the answered question: ");
+        int questionId = scanner.nextInt();
+        scanner.nextLine();
+        System.out.println("Write the text: ");
+        String text = scanner.nextLine();
+        int checkNumber = answerManagementService.addAnswer(questionId, currentUser.getUserId(), text);
+        if (checkNumber == -1) {
+            System.out.println("The selected question does not exist!");
+        } else {
+            System.out.println("The answer was added with id: " + checkNumber);
+        }
+
+    }
+
+    private void handleListAllAnswers() {
+        answerManagementService.listAllAnswers().forEach(s -> System.out.println(s.toString()));
+    }
+
+    private void handleDeleteAnswer() {
+        System.out.println("Give the answer id you want to delete: ");
+        int answerId = scanner.nextInt();
+        scanner.nextLine();
+        int checkNumber = answerManagementService.deleteAnswer(answerId, currentUser.getUserId());
+        if (checkNumber == -1) {
```

Same as above, with the addition that you should throw an exception if the user may not edit / delete this answer.